inference-engine/tests/functional/plugin/cpu/single_layer_tests/psroi_pooling.cpp

```diff
@@ -0,0 +1,182 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "test_utils/cpu_test_utils.hpp"
+
+#include "ngraph_functions/builders.hpp"
+#include "ngraph_functions/utils/ngraph_helpers.hpp"
+
+using namespace InferenceEngine;
+using namespace CPUTestUtils;
+
+namespace CPULayerTestsDefinitions {
+namespace {
+    std::vector<float> proposal;
+    std::vector<size_t> featureMapShape;
+    size_t spatialBinsX;
+    size_t spatialBinsY;
+    float spatialScale;
+    size_t groupSize;
+    size_t outputDim;
+    std::string mode;
+}  // namespace
+
+typedef std::tuple<
+        std::vector<size_t>,            // feature map shape
+        std::vector<float>,             // coords shape
+        size_t,                         // output_dim
+        size_t,                         // group_size
+        float,                          // Spatial scale
+        size_t,                         // spatial_bins_x
+        size_t,                         // spatial_bins_y
+        std::string                     // mode
+> PSROIPoolingSpecificParams;
+
+typedef std::tuple<
+        PSROIPoolingSpecificParams,
+        InferenceEngine::Precision,     // Net precision
+        LayerTestsUtils::TargetDevice   // Device name
+> PSROIPoolingLayerTestParams;
+
+typedef std::tuple<
+        CPULayerTestsDefinitions::PSROIPoolingLayerTestParams,
+        CPUSpecificParams> PSROIPoolingLayerCPUTestParamsSet;
+
+class PSROIPoolingLayerCPUTest : public testing::WithParamInterface<PSROIPoolingLayerCPUTestParamsSet>,
+                             virtual public LayerTestsUtils::LayerTestsCommon, public CPUTestsBase {
+public:
+    static std::string getTestCaseName(testing::TestParamInfo<PSROIPoolingLayerCPUTestParamsSet> obj) {
+        CPULayerTestsDefinitions::PSROIPoolingLayerTestParams basicParamsSet;
+        CPUSpecificParams cpuParams;
+        std::tie(basicParamsSet, cpuParams) = obj.param;
+        std::string td;
+        Precision netPr;
+        PSROIPoolingSpecificParams psroiPar;
+        std::tie(psroiPar, netPr, td) = basicParamsSet;
+        std::tie(featureMapShape, proposal, outputDim, groupSize,
+                 spatialScale, spatialBinsX, spatialBinsY, mode) = psroiPar;
+        std::ostringstream result;
+        result << "ROIAlignTest_";
+        result << std::to_string(obj.index) << "_";
+        result << "binsX=" << spatialBinsX << "_";
+        result << "binsY=" << spatialBinsY << "_";
+        result << "spatialScale=" << spatialScale << "_";
+        result << "outputD=" << outputDim << "_";
+        result << "groupS=" << groupSize << "_";
+        result << netPr.name() << "_";
+        result << mode << "_";
+        result << CPUTestsBase::getTestCaseName(cpuParams);
+        return result.str();
+    }
+protected:
+    void SetUp() override {
```

Comment:
As far as I can see shared PSROIPooling test uses `fillROITensor` routine to create input data but this one doesn't. What's the reason?

Reply:
It caused by bf16 input cases. Shared tests uses `makeParams(prc, {in0, in1})` syntax with common precision to create params. `fillROITensor` fills second element of params vector in `Infer()`method. I cannot use such syntax, because of bf16 - fp32 case for `featureMap` and `rois`. So, I cannot reuse `fillROITensor` inside `Infer()`. Moreover, I pay more attention to the values of proposal, because of possible failings of PSROI in average mode with non-fractional coordinates.